### ISSUE TITLE
make a model for facial emotion detection

### ISSUE THREAD

**Issue by TAHIR0110** [TAHIR0110/ThereForYou#26]
Develop a model to detect facial emotions from images or video frames. Gather diverse data, train the model, and assess its accuracy.

**Comment by utsabsamadder** [comment]
Hello, I have a good knowledge of Deep Learning. I used Pytorch in a Project of Handwriting Analysis. Recently I used YOLO to do a project on Pothole Detection. I would like to be a part of this project to enhance it even more. 

Thanks and Regards.

**Comment by gratusrichard** [comment]
@TAHIR0110  could I get assigned to this under gssoc? I believe this could implemented with #21 and could be integrated in such a way that the picture could be sent to the end point and receive a response with the result. I can use a face landmark detector to detect facial expressions and assign emotions to it.

**Comment by Kyouma45** [comment]
@TAHIR0110  I am interested in working on this. Please assign this issue to me

**Comment by Atharv714** [comment]
@utsabsamadder @Kyouma45 @gratusrichard done

**Comment by hit-haa18** [comment]
@TAHIR0110 I would like to contribute to this issue under GSSOC'24 could you please assign this to me

**Comment by TAHIR0110** [comment]
kindly write the approach as well. @akv2011 @utsabsamadder @hit-haa18 how and what you will use to build the model.and what Evaluation Metric you would use to identify the accuracy.

**Comment by hit-haa18** [comment]
My approach would be :
 Gather a diverse dataset of facial images or video frames labeled with different emotions (e.g., happy, sad, angry, surprised). Ensure the dataset represents various expressions, lighting conditions, and backgrounds to improve model robustness.Augment the training dataset with transformations like rotation, scaling, flipping, and cropping to increase model robustness and generalize better to unseen data.
Try out different pre-trained models like VGG16,ResNet,Inception and fine-tune the model and choose the one with highest accuracy (or even build my own CNN model).
Evalution metrics I would use to identify accuracy are F1-score and confusion matrix.

**Comment by akv2011** [comment]
> kindly write the approach as well. @akv2011 @utsabsamadder @hit-haa18 how and what you will use to build the model.and what Evaluation Metric you would use to identify the accuracy.

Dataset : Example
-->https://www.kaggle.com/datasets/muhammadhananasghar/human-emotions-datasethes/data
![Screenshot from 2024-05-12 16-51-05](https://github.com/TAHIR0110/ThereForYou/assets/137097109/5a30411d-9c19-46a9-8428-42c36238924b)

Optimization: Using TensorFlow to run optimal level of parallelism (number of threads)  during data augmentation, maximizing performance.

Loss Function using Categorical Crossentropy

Choosing best Model based On F1 score
Hugging face : ViT (Vision Transformer)
Model : MobileNet,VGG ,ResNet, GoogLeNet

**Comment by TAHIR0110** [comment]
@hit-haa18 assigned!!

**Comment by TAHIR0110** [comment]
@akv2011 I'll add more issues like this only, I'd love if you could work in those.

**Comment by akv2011** [comment]
> @akv2011 I'll add more issues like this only, I'd love if you could work in those.

Will do Thanks for considering

**Comment by hit-haa18** [comment]
> @hit-haa18 assigned!!

Thankyou

**Comment by hit-haa18** [comment]
@TAHIR0110 could you give me a deadline for this as I have an exam comping up
It will be really helpful

**Comment by akv2011** [comment]
> @akv2011 I'll add more issues like this only, I'd love if you could work in those.

@TAHIR0110 If possible feel free to assign any similar issues like this ... Thank you

**Comment by tejaswini4996** [comment]
I would love contributuong to this issue will you please asign it to me?
